### DISCUSSION TITLE
A marker/ pen tool to write on the pdf would be very appriciated

### DISCUSSION THREAD

**Discussion by surajsahu12** [sumatrapdfreader/sumatrapdf#2940] [Ideas]
A marker/ pen tool to write on the pdf would be wonderful, as it would provide a way to write short memos. Similar to the tools one gets when using snipping tool

**Comment by GitHubRulesOK** [comment]
this is currently available in your system via Edge. but may be limited to files under 32MB see https://github.com/sumatrapdfreader/sumatrapdf/issues/1959#issuecomment-864493199 image 
![122660925-4c878880-d17d-11eb-973b-851788fcc8aa 1](https://user-images.githubusercontent.com/12754913/188850117-1e44542d-9e20-46ec-afd9-a63b16362bb3.png)

For how to open the file in the inking window see https://github.com/sumatrapdfreader/sumatrapdf/discussions/2454#discussioncomment-2500800 

note the file should be on local drive must be saved as same name for SumatraPDF to auto reload as changed.
issue is covered by https://github.com/sumatrapdfreader/sumatrapdf/issues/961 and others

**Reply by surajsahu12** [reply]
Thankyou for the reply, I'll be using edge for this function. From the threads you mentioned i figured out it would be a time consuming task, but i do hope that someday, writing on the pdf would be available in sumatra (although its only a reader app, a wonderful one). 

have a good day.

**Reply by GitHubRulesOK** [reply]
Edge has inking, variable free text and Read Aloud so better for those tasks as a second window for current file.

The intention for SumatraPDF was to extend to inking but its a step after several other necessary steps, and if you try inking in MuPDF-GL.exe you can see some of the challenges in porting that across.
